Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,293 @@
+            public override IReadOnlyList<byte> DataBytes => Array.Empty<byte>();
+            public override int Length => 1;
+        }
+
+        public sealed class DutyCycleRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.DutyCycleCmd;
+            public override DutyCycleRequest Subject => new DutyCycleRequest(3);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 3 };
+            public override int Length => 2;
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest<DutyCycleRequest>(actual => actual.Cid == Cid
+                                                             && actual.DutyCyclePL == Subject.DutyCyclePL,
+                                                      @"{""cid"":4,""dutyCyclePL"":3}");
+        }
+
+        public sealed class LinkAdrAnswerTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.LinkADRCmd;
+            public override LinkADRAnswer Subject => new LinkADRAnswer(1, true, false);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 0b110 };
+            public override int Length => 2;
+
+            [Fact]
+            public void FromBytes_Success() => FromBytesTest(actual => actual.Cid == Cid
+                                                                    && actual.PowerAck
+                                                                    && actual.DRAck
+                                                                    && !actual.CHMaskAck,
+                                                             bytes => new LinkADRAnswer(new ReadOnlySpan<byte>(bytes)),
+                                                             DataBytes);
+        }
+
+        public sealed class LinkAdrRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.LinkADRCmd;
+            public override LinkADRRequest Subject => new LinkADRRequest(1, 2, 3, 4, 5);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 0b10010, 0b11, 0, 0b1000101 };
+            public override int Length => 5;
+
+            private Predicate<LinkADRRequest> Assert => actual => actual.Cid == Cid
+                                                               && actual.DataRate == Subject.DataRate
+                                                               && actual.TxPower == Subject.TxPower
+                                                               && actual.ChMask == Subject.ChMask
+                                                               && actual.ChMaskCntl == Subject.ChMaskCntl
+                                                               && actual.NbRep == Subject.NbRep;
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest(Assert, @"{""cid"":3,""dataRate"":1,""txPower"":2,""chMask"":3,""chMaskCntl"":4,""nbRep"":5}");
```

Comment:
nit: If `DeserializationTest` called `JsonUtil.Strictify` on the JSON, then all the JSON text could be easier to read (by not needing quoted JSON object members names).

Reply:
Done in c6fa017c9295979c98b31f8f3a9c954a3cf56b99